downloads/factories.py

```diff
@@ -0,0 +1,120 @@
+from urllib.parse import urljoin
+
+import factory
+import requests
+
+from users.factories import UserFactory
+
+from .models import OS, Release, ReleaseFile
+
+
+class OSFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = OS
+        django_get_or_create = ('slug',)
+
+    creator = factory.SubFactory(UserFactory)
+
+
+class ReleaseFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = Release
+        django_get_or_create = ('slug',)
+
+    creator = factory.SubFactory(UserFactory)
+    is_published = True
+
+
+class ReleaseFileFactory(factory.DjangoModelFactory):
+
+    class Meta:
+        model = ReleaseFile
+        django_get_or_create = ('slug',)
+
+    creator = factory.SubFactory(UserFactory)
+    release = factory.SubFactory(ReleaseFactory)
+    os = factory.SubFactory(OSFactory)
+
+
+class APISession(requests.Session):
+    base_url = 'https://www.python.org/api/v2/'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.headers.update(
+            {
+                "Accept": "application/json",
+                "User-agent": "pythondotorg/create_initial_data ({})".format(
+                    requests.utils.default_user_agent()
+                ),
+            }
+        )
+
+    def request(self, method, url, **kwargs):
+        url = urljoin(self.base_url, url)
+        response = super().request(method, url, **kwargs)
+        response.raise_for_status()
+        return response
+
+
+def _get_id(obj, key):
+    """
+    Get the ID of an object by extracting it from the resource uri.
+    """
+    return (obj.pop(key, '') or '').rstrip('/').rpartition('/')[-1]
+
+
+def initial_data():
+    """
+    Create the data for the downloads section by importing
+    it from the python.org API.
+
+    """
+    objects = {
+        'oss': {},
+        'releases': {},
+        'release_files': {},
+    }
+
+    with APISession() as session:
+        for key, resource_uri in [
+            ('oss', 'downloads/os/'),
+            ('releases', 'downloads/release/'),
+            ('release_files', 'downloads/release_file/')
+        ]:
+            response = session.get(resource_uri)
+            object_list = response.json()
+
+            for obj in object_list:
+                objects[key][_get_id(obj, 'resource_uri')] = obj
+
+    # Create the list of operating systems
+    objects['oss'] = {k: OSFactory(**obj) for k, obj in objects['oss'].items()}
+
+    # Create all the releases
+    for key, obj in objects['releases'].items():
+        obj.pop('release_page')  # Ignore release pages
```

Comment:
Did you skip adding release pages for now or did you get blocked by something? I think we can add a `# TODO` comment if it takes too long to implement it.

Reply:
I think I got blocked on something, but I cannot remember what. I'll add a `TODO` for now

Reply:
Great! Feel free to file an issue if you remember what was the blocker.